src/main/java/school/hei/haapi/service/UserService.java

```diff
@@ -233,4 +242,38 @@ public Integer getStudentsAlternatingSize(
         .toList()
         .size();
   }
+
+  public Boolean computeIsRepeatingThisYear(User user) {
+    Set<school.hei.haapi.model.Promotion> distinctPromotions = getStudentDistinctPromotions(user);
+    if (distinctPromotions.isEmpty()) {
+      return false;
+    }
+    school.hei.haapi.model.Promotion mostRecentPromotion =
+        distinctPromotions.stream()
+            .sorted(Comparator.comparingInt(prom -> prom.getStartDate().getYear()))
+            .toList()
+            .getLast();
+    return didRepeatYear(user)
+        && (mostRecentPromotion.getStartDate().getYear() == Year.now().getValue()
+            || mostRecentPromotion.getStartDate().getYear() + 1 == Year.now().getValue());
```

Comment:
why you add + 1 when you calculate it ?

Reply:
Suppose current year is 2023 and the most recent promotion start date year is 2023, how do I know if a student repeat a year ? I confirm that he has at least 2 prom, then if the most recent Promotion's start date Year == this year (2023) then the student repeated a year, 
but what if the current year is 2024 ?
Promotion's start date Year + 1 == this year (2023+1 == 2024) then the student repeated a year because a year in college is like this 2023-2024
Does it touch the promotion 2024-2025 ?
if current year is 2023, then it's not but if current year is 2024, then I just need to add a condition that the prom start year + 1 (2025 here) <= current year